build/vsts-ci.yml

```diff
@@ -226,21 +226,20 @@ phases:
       msbuildVersion: 15.0
     continueOnError: false
 
-  - task: NuGetCommand@2 
```

Comment:
I asked @codemzs and he told me he was okay with me removing this task that publishes the nugets into the private feed. Still, he was concerned about not knowing if this would "break any dependencies". It seems it doesn't, but if anyone has a reason for not removing this task, please tell me. (@eerhardt ?).

Reply:
So after I mentioned to @codemzs that publishing to the public feed might not always automatically work because of problems on the side of AzureDevOps, he recommended me to put back the task that publishes to the private feed.

Still, I decided to put it back after the other publishing tasks, since if the task of publishing to the private feed fails it would interrupt the other publishing tasks.

If you want to see it before the task of publishing to the public feed, please tell me.

Reply:
> So after I mentioned to @codemzs that publishing to the public feed might not always automatically work because of problems on the side of AzureDevOps

This is not publishing to the public feed itself, the build definition already has that flakiness because of network connection issues or other unrelated issues to publishing... I think we shouldn't be publishing to both sides as it just slows down builds, it's expensive and nobody will be looking at the internal feed.

Reply:
I agree that the problem isn't in the publishing step itself.

I've looked into the builds that were executed in the private pipelines before this PR, and it seems to me that in many cases the nugets weren't even published to the private feed because something went wrong with the pipeline.

After this PR got merged, the automatic build failed at one of the steps that runs before even trying to publish the nugets to the public feed. After Maryam's latest PR got merged an hour later, the automatic build failed while installing a plugin even before building the Windows x86 and Windows x64 versions. I've also tried to manually rerun the builds, but I also get similar problems, and even in one occasion the build failed because it spent over an hour trying to build and sign the Windows x64 binaries. So, so far, the things I've introduced in this PR haven't been actually executed in Azure DevOps in the master branch (I've only tested my changes by running them from a branch in the private repo, and that worked as expected and published the nugets to the feed, but even then it required me to run a couple of manual builds because of the same problems I've already described).

I am somewhat worried about all of these, because it seems that other people will rely on the public feed to be always up-to-date (like @frank-dong-ms nightly build tests). But I wouldn't know if there could be other options, or workarounds to all of this.

Reply:
Ok, so in the last two builds (one manual, and one triggered automatically after @codemzs latest PR), the nugets were published successfully to the public feed, but the pipeline actually failed when publishing to the private feed.

Reply:
About that last point, it actually seems that the nugets were correctly published to the feed, but the feed decides (for whatever reason) to select previous published nugets and show them directly in the feed instead of the latest ones. If I go to the 'versions' tab of an specific nugget, I actually can see that it got updated on the builds the last builds, but it is selecting to show the one built 8 hours ago: 
https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=MachineLearning&package=Microsoft.Extensions.ML&protocolType=NuGet&version=1.4.0-preview3-28229-9&view=versions

I am guessing this is, for some reason, the expected behavior, and that I was wrong when saying that the Feed takes time to reflect the changes... it just doesn't display the latest version on the feed.

Reply:
Let's only have a single feed - the public feed. Having 2 feeds will only increase confusion, storage, and increase the chance of failure.